server/routes/api.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express')
+const urllib = require('urllib')
+const router = express.Router()
+
+const _teamToIDs = {
+    "lakers": "1610612747",
+    "warriors": "1610612744",
+    "heat": "1610612748",
+    "suns": "1610612756"
+}
+
+let _dreamTeam = []
+let _dreamPlayers = {}
+
+const _myTeams = {
+    "My teams": null
+
+}
+
+const _newDreamTeam = () => {
+    _dreamTeam = []
+    _dreamPlayers = {}
+}
+
+const _ifPlayerIsInDreamTeam = (name) => {
+    if (_dreamPlayers[name] === undefined || _dreamPlayers[name] === false) {
+        _dreamPlayers[name] = true
+        return true
+    }
+    return false
+}
+const _removeFromDreamTeam = (name) => {
+    _dreamPlayers[name] = false
+    _dreamTeam.splice(_dreamTeam.findIndex((player) => `${player.firstName} ${player.lastName}` === name), 1)
+}
+const _getDreamTeam = () => { return { roster: _dreamTeam } }
+
+router.get('/teams/:teamName', function (request, response) {
+    const { teamName } = request.params
+    urllib.request('http://data.nba.net/10s/prod/v1/2017/players.json', (err, _response) => {
+        if (!err)
+            response.send({
+                roster: JSON.parse(_response).league.standard.filter(player => player.isActive && player.teamId === _teamToIDs[teamName]).map(player => {
```

Comment:
Don't do this in the response. Better to do your filtering and mapping and store it in a variable and then send that variable. Cleaner that way.

Good job on the chaining!